src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * @author leslieharland
+ */
+
+package duke.task;
+
+import duke.command.Operation;
+
+import java.time.format.DateTimeFormatter;
+
+/**
+ * The Class Task.
+ */
+public abstract class Task implements TaskPrintable, Parseable {
+
+    protected String description;
+
+    protected boolean isDone;
+
+    protected String type;
+
+    protected String userKeyedDateString;
+
+    protected java.time.LocalDate datetime;
+
+    /**
+     * Instantiates a new task.
+     *
+     * @param description the description
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * This constructor is used to retain most useful properties needed to create
+     * task at its core.
+     *
+     * @param description the description
+     */
+    public Task(boolean isDone, String description, String type) {
+        this.isDone = isDone;
+        this.description = description;
+        this.type = type;
+    }
+
+    /**
+     * Instantiates a new task.
+     *
+     * @param description the description
+     * @param datetime    the datetime
+     */
+    public Task(String description, String datetime) {
+        this.userKeyedDateString = datetime;
+        setDatetime(datetime);
+        this.description = description;
+    }
+
+    /**
+     * Instantiates a new task.
+     *
+     * @param isDone      the is done
```

Comment:
I think there should only be 1 space between @param isDone and the is done, and that the description for this parameter could be more detailed in explaining perhaps what is done.